src/sagemaker/s3.py

```diff
@@ -153,4 +153,8 @@ def list(s3_uri, session=None):
         bucket, key_prefix = parse_s3_url(url=s3_uri)
 
         file_keys = sagemaker_session.list_s3_files(bucket=bucket, key_prefix=key_prefix)
-        return [os.path.join("s3://", bucket, file_key) for file_key in file_keys]
+        path = ["s3://", bucket]
+        for file_key in file_keys:
+            path.append(os.sep)
+            path.append(file_key)
+        return "".join(path)
```

Comment:
using `os.sep` doesn't actually address the issue, since `os.sep` is `'\'` on Windows. For a solution compatible with both Python 2 and 3, you're better off using a format string.

See https://github.com/aws/sagemaker-python-sdk/pull/1439#discussion_r414775571